smart_open/tests/test_smart_open.py

```diff
@@ -486,6 +485,19 @@ def test_s3_mode_mock(self, mock_write, mock_boto):
         mock_boto.connect_s3().get_bucket.assert_called_with("mybucket")
         self.assertTrue(mock_write.called)
 
+    @mock.patch('smart_open.smart_open_lib.subprocess')
+    def test_hdfs(self, mock_subprocess):
+        """Is HDFS write called correctly"""
+        smart_open_object = smart_open.HdfsOpenWrite(smart_open.ParseUri("hdfs:///tmp/test.txt"))
```

Comment:
It is a good idea to add these tests. But IMO it's better to leave this pull request focus on the write operation alone, as it was from the beginning and add new tests for smart_open.smart_open on a separate pull request.